test/suite/test_timestamp04.py

```diff
@@ -44,103 +44,244 @@ def timestamp_ret_str(t):
     return s
 
 class test_timestamp04(wttest.WiredTigerTestCase, suite_subprocess):
-    tablename = 'test_timestamp04'
-    uri = 'table:' + tablename
+    table_ts_log     = 'ts04_ts_logged'
+    table_ts_nolog   = 'ts04_ts_nologged'
+    table_nots_log   = 'ts04_nots_logged'
+    table_nots_nolog = 'ts04_nots_nologged'
+
+    # Below two lists conncfg and types are not removed but kept commented for
+    # better understanding of the combinations of scenarios.
+    # Different scenarios are manaually provided instead of combinations as
+    # the cache_size for lsm minimum requirement is 31MB.
+    #conncfg = [
+        #('nolog', dict(conn_config='create,cache_size=20MB', using_log=False)),
+        #('V1', dict(conn_config='create,cache_size=20MB,log=(enabled),compatibility=(release="2.9")', using_log=True)),
+        #('V2', dict(conn_config='create,cache_size=20MB,log=(enabled)', using_log=True)),
+    #]
+
+    #types = [
+        #('col_fix', dict(empty=1, uri='table:', extra_config=',key_format=r, value_format=8t',is_lsm=False)),
+        #('col_var', dict(empty=0, uri='table:', extra_config=',key_format=r',is_lsm=False)),
+        #('lsm', dict(empty=0, uri='lsm:', extra_config=',type=lsm',is_lsm=True)),
+        #('row', dict(empty=0, uri='table:', extra_config='',is_lsm=False)),
+    #]
 
     scenarios = make_scenarios([
```

Comment:
test_compact02.py seems to utilise overriding standard connection open methods to create connections for several scenarios based on different cache sizes. Can we try something like that here in this test.

Reply:
Done.